Libs/DICOM/Widgets/ctkDICOMBrowser.cpp

```diff
@@ -657,3 +667,168 @@ void ctkDICOMBrowser::onModelSelected(const QItemSelection &item1, const QItemSe
   Q_D(ctkDICOMBrowser);
   d->ActionRemove->setEnabled(true);
 }
+
+//----------------------------------------------------------------------------
+void ctkDICOMBrowser::addSelectionLabelsToContextMenu(QStringList uids, QMenu *menu)
+{
+  Q_D(ctkDICOMBrowser);
+
+  if (!menu || uids.isEmpty())
+    {
+    return;
+    }
+
+  // add non clickable labels to the menu
+  QLabel *headerLabel = new QLabel(tr("Selected:"), menu);
+  QWidgetAction *headerAction = new QWidgetAction(menu);
+  headerAction->setDefaultWidget(headerLabel);
+  menu->addAction(headerAction);
+
+  int numUIDs = uids.size();
+  for (int i = 0; i < numUIDs; ++i)
+    {
+    QString uid = uids.at(i);
+
+    // try using the given UID to find a descriptive string
+    QString patientName = d->DICOMDatabase->nameForPatient(uid);
+    QString studyDescription = d->DICOMDatabase->descriptionForStudy(uid);
+    QString seriesDescription = d->DICOMDatabase->descriptionForSeries(uid);
+
+    QLabel *selectionLabel;
+    if (!patientName.isEmpty())
+      {
+      selectionLabel = new QLabel(patientName);
+      }
+    else if (!studyDescription.isEmpty())
+      {
+      selectionLabel = new QLabel(studyDescription);
+      }
+    else if (!seriesDescription.isEmpty())
+      {
+      selectionLabel = new QLabel(seriesDescription);
+      }
+    if (!selectionLabel->text().isEmpty())
```

Comment:
nitpick: Could be simpler to simply set a variable `QString selectionDescription;`

That code would then become:

```
if (!selectionDescription.isEmpty())
  {
  QWidgetAction *selectionAction = new QWidgetAction(menu);
  selectionAction->setDefaultWidget(new QLabel(selectionDescription));
  menu->addAction(selectionAction);
  }
```